src/main/java/org/jvnet/hudson/test/JellyTestSuiteBuilder.java

```diff
@@ -103,11 +111,52 @@ protected void runTest() throws Exception {
             if (requirePI) {
                 ProcessingInstruction pi = dom.processingInstruction("jelly");
                 if (pi == null || !pi.getText().contains("escape-by-default")) {
-                    throw new AssertionError("<?jelly escape-by-default='true'?> is missing in "+jelly);
+                    errors.add("<?jelly escape-by-default='true'?> is missing in "+jelly);
                 }
-
             }
             // TODO: what else can we check statically? use of taglibs?
+            checkScriptElement(dom);
+            checkJavaScriptAttributes(dom);
+            if (!errors.isEmpty()) {
+                String message = String.join("\n", errors);
+                throw new AssertionError(message);
+            }
+        }
+
+        private void checkJavaScriptAttributes(Document dom) {
+            List<Node> allNodes = dom.selectNodes("//*");
+            allNodes.forEach(n -> {
+                Element element = (Element) n;
+                Attribute onclick = element.attribute("onclick");
+                Attribute checkUrl = element.attribute("checkUrl");
+                Attribute checkDependsOn = element.attribute("checkDependsOn");
+                if (checkUrl != null  && checkDependsOn == null) {
+                    errors.add("Usage of 'checkUrl' without 'checkDependsOn' in "+jelly);
```

Comment:
I'm wondering if this is clear enough to people, e.g. in most cases people need to add a `checkDependsOn=""`.

Personally I think this isn't clear enough on its own.

(It would be good if this could be fixed so it doesn't need the empty checkDependsOn)

Reply:
Probably when we have CSP enabled by default and all plugins are adjusted we could remove an empty `checkDependsOn`, until then we need it to find out if we have to run the `eval` on the `checkUrl`.
We could add a link to https://www.jenkins.io/doc/developer/security/csp/ whenever there is an inline javascript found.